### PR TITLE
test: add scenario for eth sign using hardware wallet

### LINKED ISSUE #22026: Hardware wallet: Eth Sign

### PR DESCRIPTION
**Pull request by segun**
## **Description**
In this PR we add the scenario for the flow: eth sign with hardware wallet.
This task belongs to the effort of documenting manual QA flows[ in this Epic](https://github.com/MetaMask/metamask-extension/issues/21962).

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/23313?quickstart=1)

## **Related issues**

Fixes: #22026 

## **Manual testing steps**

1. Check the video below.

## **Screenshots/Recordings**
Performing the steps defined in the scenario

https://github.com/MetaMask/metamask-extension/assets/44811/a858aa5e-2bde-411c-bc4e-dd787efa778f

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [x] I've linked related issues
- [x] I've included manual testing steps
- [x] I've included screenshots/recordings if applicable
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [x] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by seaona — CHANGES_REQUESTED** [review]
See comment

**Review by blackdevelopa — DISMISSED** [review]
Looks good. The csv is formatted as expected

**Review by jpuri — APPROVED** [review]

**Review by jpuri — DISMISSED** [review]

File: test/scenarios/13. sign/eth sign with hardware wallet.csv [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+10,"Open the accounts options menu, select settings. In advanced tab, enable the eth_sign request.",,,"The ""Use at your own risk"" modal is shown",
+11,"Click the ""I understand that I can lose all of my funds and NFTs if I enable eth_sign requests."" checkbox, Click Continue button and enter the text ""I only sign what I understand"" in the provided text field, Click Enable buton",,,The eth_sign toggle is ON.,
+12,Open the test dapp in another tab.,,https://metamask.github.io/test-dapp/,,
+13,Click Connect.,,,"The MetaMask popup is opened with the Connect with MetaMask screen displayed. Your imported hardware wallet account is selected.",
+14,Click Next and Connect with the hardware wallet account.,,,"The MetaMask popup is closed. You are connected to the test dapp.",
+15,"Click Sign in the ""Eth Sign"" section",,,"The info modal instructions for signing with hardware wallet is displayed. The signature message is in Hex formatting.",
+16,"Click Sign in the displayed modal Then the ""Your funds may be at risk"" modal is shown click Sign",,,The signature hash is displayed on the test dapp.,
```
Comment (seaona) [inline test/scenarios/13. sign/eth sign with hardware wallet.csv:17]: I think here we don't expect yet that the 'Signature hash is displayed on the test dapp yet' but we do expect it in the next step 17

Comment (seaona) [reply test/scenarios/13. sign/eth sign with hardware wallet.csv:17]: the rest looks good to me!

**Review by seaona — APPROVED** [review]
LGTM

**Review by sleepytanya — APPROVED** [review]

**Pull request merged**